src/controllers/attachment/index.ts

```diff
@@ -0,0 +1,94 @@
+import { UploadedFile, Request, Controller, Get, Route, Path, Post, Header, Response, SuccessResponse } from 'tsoa'
+import { Logger } from 'pino'
+import express from 'express'
+
+import { logger } from '../../lib/logger'
+import Database, { Models, Query } from '../../lib/db'
+import type { Attachment } from '../../models'
+import { BadRequest, NotFound } from '../../lib/error-handler'
+
+type File = {
+  [k: string]: string
+}
+
+@Route('attachment')
+export class attachment extends Controller {
+  log: Logger
+  dbClient: Database = new Database()
+  db: Models<Query> | any // TMP this is the only one could not address now
+
+  constructor() {
+    super()
+    this.log = logger.child({ controller: '/attachment' })
+    // TMP will be updated with a wrapper so ORM client independant
+    this.db = this.dbClient.db()
+  }
+
+  @Get('/')
+  @SuccessResponse(200, 'returns all attachments')
+  public async get(): Promise<Attachment[]> {
+    this.log.debug('retrieving all attachments')
+    const result = await this.db.attachment()
+
+    return result.map(({ binary_blob, ...item }: Attachment) => ({
+      ...item,
+      size: binary_blob.size,
+    }))
+  }
+
+  @Post('/')
+  @SuccessResponse(201, 'attachment has been created')
+  public async create(@Request() req: express.Request, @UploadedFile() file: File): Promise<string> {
```

Comment:
Interesting so does this magically handle that it might be either a multi part form or a json body?

Reply:
yes! this is why needed to install multer as dev dep. However it did not like both type e.g. multipart/json

Reply:
how do you upload json on swagger? Only option is multipart
![image](https://user-images.githubusercontent.com/40722025/225895325-0ed4e25b-58ef-44e8-a3cd-3aad088c78f1.png)

Reply:
this is very annoying and is a regression from what we had before. Sadly it appears you're right @n3op2 and tsoa isn't capable of this based off of https://github.com/lukeautry/tsoa/issues/968. I'm honestly not sure what's best as open api does allow this behaviour and the tsoa guys have just decided it's not ok without giving reasons

Reply:
Mmm... any ideas how to move forward? As a work around I can do something like I did for `getById()` route. We can pass a header via curl and it will work. But that means that there will be two possible routes e.g.

```ts
  public async getById(
    ...
    @Header('return-type') type: 'json' | 'file'
    ...
    if (type === 'file' || accept === 'application/octet-stream') {
    return ...
```